src/main/java/picocli/CommandLine.java

```diff
@@ -11987,19 +11987,17 @@ private boolean is(ArgSpec p, String attribute, boolean value) {
                 return new ITypeConverter<Object>() {
                     @SuppressWarnings("unchecked")
                     public Object convert(String value) throws Exception {
-                        String sensitivity = "case-sensitive";
-                        if (commandSpec.parser().caseInsensitiveEnumValuesAllowed()) {
-                            String upper = value.toUpperCase();
+                        try { return Enum.valueOf((Class<Enum>) type, value); }
+                        catch (IllegalArgumentException ex) {
+                            boolean insensitive = commandSpec.parser().caseInsensitiveEnumValuesAllowed();
                             for (Object enumConstant : type.getEnumConstants()) {
-                                if (upper.equals(String.valueOf(enumConstant).toUpperCase())) { return enumConstant; }
+                                String thisName = enumConstant.toString();
+                                if( value.equals(thisName) || insensitive && value.equalsIgnoreCase(thisName) ) { return enumConstant; }
```

Comment:
Let's add a bugfix to also compare `enumConstant.name()` case-insensitively:

```java
String str = enumConstant.toString();
String name = enumConstant.name();
if (value.equals(str) || value.equals(name) || insensitive && (value.equalsIgnoreCase(str) || value.equalsIgnoreCase(name))) {
    return enumConstant;
}
```

Please keep the whitespace conventions the same as the original. :-) So, `if (xxx)` instead of `if( xxx )`.